### ISSUE TITLE
Fix false positives for Less dynamic functions in unit-no-unknown

### ISSUE THREAD

**Issue by vilchik-elena** [stylelint/stylelint#3721]
> Clearly describe the bug

There are FPs on `;` for rule `unit-no-unknown` on `less` files, for example here https://github.com/ant-design/ant-design/blob/master/components/style/color/bezierEasing.less#L4

> Which rule, if any, is the bug related to?

`unit-no-unknown`

> What CSS is needed to reproduce the bug?

```less
.bezierEasingMixin() {
@functions: ~`(function() {
  var NEWTON_ITERATIONS = 4;
  })`
}
```

> What stylelint configuration is needed to reproduce the bug?
```json
{
  "rules": {
    "unit-no-unknown": "true"
  }
}
```

> Which version of stylelint are you using?

`9.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to Less

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
test.css
3:4    Unexpected unknown unit ";" (unit-no-unknown)
```

**Comment by jeddy3** [comment]
@vilchik-elena Thanks for the report and for using the template.

I've not seen that construct before. Is it a documented feature of Less?

You can also try `master` as [it now includes](https://github.com/stylelint/stylelint/pull/3687) the latest version of the Less parser.

**Comment by vilchik-elena** [comment]
Indeed, that's a JS inside less
